src/org/rust/lang/core/psi/impl/mixin/RustPathPartImplMixin.kt

```diff
@@ -24,6 +25,30 @@ abstract class RustPathPartImplMixin(node: ASTNode) : RustNamedElementImpl(node)
             if (it.firstChild != null) it else null
         }
 
+    private val isViewPath: Boolean
+        get() = when (parent) {
+            is RustViewPath          -> true
+            is RustPathPartImplMixin -> (parent as RustPathPartImplMixin).isViewPath
+            else                     -> false
+        }
+
+    /**
+     *  Returns `true` if this is a first segment of a fully qualified path.
+     *
+     *  Paths in use items are special, they are implicitly FQ.
+     *
+     *  Example:
+     *
+     *    ```Rust
+     *    use ::foo::bar;   // FQ
+     *    use foo::bar;     // FQ, the same as the above
+     *
+     *    fn main() {
+     *        ::foo::bar;   // FQ
+     *        foo::bar;     // not FQ
+     *    }
+     *    ```
+     */
     override val isFullyQualified: Boolean
-        get() = getQualifier() == null && getSeparator() != null
+        get() = getQualifier() == null && (getSeparator() != null || (isViewPath && self == null && `super` == null))
```

Comment:
This `isFullyQualified` would return false for (the full-path) `::foo::bar`, while it shouldn't.